source/components/inputs/signature/signature.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, Optional, Input, AfterViewInit, AfterViewChecked, OnChanges, ViewChild, SimpleChange } from '@angular/core';
+import { Observable } from 'rxjs';
+import '../../../../libraries/jSignature/jSignature.min';
+
+import { services } from 'typescript-angular-utilities';
+import __object = services.object;
+import __array = services.array;
+import __guid = services.guid;
+
+import { ValidatedInputComponent, validationInputs, baseOutputs, IInputChanges } from '../validationInput';
+import { ComponentValidator } from '../../../services/componentValidator/componentValidator.service';
+import { FormComponent } from '../../form/form';
+import { ButtonComponent } from '../../buttons/index';
+import { JQueryProvider } from '../../../services/jquery/jquery.provider';
+import { emptySignature } from './emptySignature';
+
+export interface ISignatureChanges extends IInputChanges {
+	disabled: SimpleChange;
+}
+
+@Component({
+	selector: 'rlSignature',
+	template: require('./signature.html'),
+	inputs: validationInputs,
+	outputs: baseOutputs,
+	providers: [ComponentValidator],
+	directives: [ButtonComponent],
+})
+export class SignatureComponent extends ValidatedInputComponent<string> implements AfterViewInit, AfterViewChecked, OnChanges {
+	private _canvas: JQuery;
+	@ViewChild('canvas') set canvas(value: JQuery) {
```

Comment:
A little nitpicky, but I had to do a double take on this line. It might be more readable if the `@ViewChild` is on its own line above the property?